Applications/Spire/Source/Ui/TextAreaBox.cpp

```diff
@@ -0,0 +1,634 @@
+    m_submission = m_model->get_current();
+    m_submit_signal(m_submission);
+  } else if(event->type() == QEvent::Wheel) {
+    if(m_text_edit->isReadOnly()) {
+      return true;
+    }
+  }
+  return QWidget::eventFilter(watched, event);
+}
+
+void TextAreaBox::mousePressEvent(QMouseEvent* event) {
+  m_text_edit->setFocus();
+  QWidget::mousePressEvent(event);
+}
+
+void TextAreaBox::resizeEvent(QResizeEvent* event) {
+  update_layout();
+  QWidget::resizeEvent(event);
+}
+
+void TextAreaBox::commit_placeholder_style() {
+  auto alignment = m_placeholder_styles.m_alignment.value_or(
+    Qt::Alignment(Qt::AlignmentFlag::AlignLeft));
+  m_placeholder->set_alignment(alignment);
+  auto font = m_placeholder_styles.m_font.value_or(QFont());
+  if(m_placeholder_styles.m_size) {
+    font.setPixelSize(*m_placeholder_styles.m_size);
+  }
+  m_placeholder->setFont(font);
+  m_placeholder->set_text_color(m_placeholder_styles.m_color.value_or(
+    QColor()));
+  update_placeholder_text();
+}
+
+void TextAreaBox::commit_style() {
+  auto stylesheet = QString(
+    R"(QTextEdit {
+      background: transparent;
+      border-width: 0px;)");
+  m_text_edit_styles.m_styles.write(stylesheet);
+  auto alignment = m_text_edit_styles.m_alignment.value_or(
+    Qt::Alignment(Qt::AlignmentFlag::AlignLeft));
+  if(alignment != m_text_edit->alignment()) {
+    update_text_alignment();
+  }
+  auto font = m_text_edit_styles.m_font.value_or(QFont());
+  if(m_text_edit_styles.m_size) {
+    font.setPixelSize(*m_text_edit_styles.m_size);
+  }
+  m_text_edit->setFont(font);
+  if(m_text_edit_styles.m_line_height &&
+      (font.pixelSize() * *m_text_edit_styles.m_line_height) !=
+      m_computed_line_height) {
+    update_line_height();
+  }
+  if(stylesheet != m_text_edit->styleSheet()) {
+    m_text_edit->setStyleSheet(stylesheet);
+    m_text_edit->style()->unpolish(this);
+    m_text_edit->style()->polish(this);
+  }
+}
+
+bool TextAreaBox::is_placeholder_shown() const {
+  return !is_read_only() && m_model->get_current().isEmpty() &&
+    !m_placeholder_text.isEmpty();
+}
+
+QSize TextAreaBox::get_border_size() const {
+  auto borders = m_text_edit_styles.m_border_sizes;
+  return {borders.left() + borders.right(), borders.top() + borders.bottom()};
+}
+
+QSize TextAreaBox::get_padding_size() const {
+  auto padding = m_text_edit_styles.m_padding;
+  return {padding.left() + padding.right(), padding.top() + padding.bottom()};
+}
+
+void TextAreaBox::update_display_text() {
+  if(is_read_only()) {
+    if(m_text_edit->toPlainText() != m_model->get_current()) {
+      m_text_edit->blockSignals(true);
+      m_text_edit->setText(m_model->get_current());
+      m_text_edit->blockSignals(false);
+    }
+    auto line_count = std::floor(static_cast<double>((height() -
+      get_padding_size().height())) /
+      static_cast<double>(m_computed_line_height));
+    if(line_count > 0) {
+      auto lines = [&] {
```

Comment:
Remove the IIFE, it's not serving any purpose, line 360 completely invalidates the use of an IIFE.